src/DualAzureStoreFactory.cs

```diff
@@ -107,33 +108,22 @@ public async Task<IReadOnlyList<long>> GetAccountsAsync(CancellationToken cancel
         }
 
         /// <summary> Retrieve all accounts that have blobs in stores from this container. </summary>
-        private async Task<List<long>> GetAccountsAsync(CloudBlobContainer cbc, CancellationToken cancel)
+        private async Task<List<long>> GetAccountsAsync(BlobContainerClient cbc, CancellationToken cancel)
         {
             var accounts = new List<long>();
 
-            var token = default(BlobContinuationToken);
+            var result = await cbc.GetBlobsByHierarchyAsync(
+                traits: BlobTraits.All,
+                states: BlobStates.None,
+                prefix: "",
+                cancellationToken: cancel).ToListAsync().ConfigureAwait(false);
```

Comment:
We routinely manipulate hundreds of millions of blobs with this library.

`.ToListAsync()` will result in creating a list of all the blobs, which will probably not fit in memory. The previous version (based on `ListBlobsSegmentedAsync`) discarded all the blobs of a segment before requesting a new segment, and this behavior should be retained.